website/public/js/services/userServices.js

```diff
@@ -0,0 +1,416 @@
+        },
+
+        sleep: function () {
+          callOpsFunctionAndRequest('sleep', 'sleep', "POST");
+        },
+
+        online: function (status) {
+          if (status===true) {
+            settings.online = true;
+            // syncQueue();
+          } else {
+            settings.online = false;
+          };
+        },
+
+        authenticate: function (uuid, token, cb) {
+          if (!!uuid && !!token) {
+            var offset = moment().zone(); // eg, 240 - this will be converted on server as -(offset/60)
+            $http.defaults.headers.common['x-api-user'] = uuid;
+            $http.defaults.headers.common['x-api-key'] = token;
+            $http.defaults.headers.common['x-user-timezoneOffset'] = offset;
+            authenticated = true;
+            settings.auth.apiId = uuid;
+            settings.auth.apiToken = token;
+            settings.online = true;
+            if (user && user._v) user._v--; // shortcut to always fetch new updates on page reload
+            userServices.log({}, function(){
+              // If they don't have timezone, set it
+              if (user.preferences.timezoneOffset !== offset)
+                userServices.set({'preferences.timezoneOffset': offset});
+              cb && cb();
+            });
+          } else {
+            alert('Please enter your ID and Token in settings.')
+          }
+        },
+
+        authenticated: function(){
+          return this.settings.auth.apiId !== "";
+        },
+
+        getBalanceInGems: function() {
+          var balance = user.balance || 0;
+          return balance * 4;
+        },
+
+        log: function (action, cb) {
+          //push by one buy one if an array passed in.
+          if (_.isArray(action)) {
+            action.forEach(function (a) {
+              settings.sync.queue.push(a);
+            });
+          } else {
+            settings.sync.queue.push(action);
+          }
+
+          save();
+          // syncQueue(cb);
+        },
+
+        sync: function(){
+          user._v--;
+          userServices.log({});
+          sync();
+        },
+
+        save: save,
+
+        settings: settings
+      };
+
+      //load settings if we have them
+      if (localStorage.getItem(STORAGE_SETTINGS_ID)) {
+        //use extend here to make sure we keep object reference in other angular controllers
+        _.extend(settings, JSON.parse(localStorage.getItem(STORAGE_SETTINGS_ID)));
+
+        //if settings were saved while fetch was in process reset the flag.
+        settings.fetching = false;
+      //create and load if not
+      } else {
+        localStorage.setItem(STORAGE_SETTINGS_ID, JSON.stringify(defaultSettings));
+        _.extend(settings, defaultSettings);
+      }
+
+      //If user does not have ApiID that forward him to settings.
+      if (!settings.auth.apiId || !settings.auth.apiToken) {
+
+        if (MOBILE_APP) {
```

Comment:
this IF can be removed, it was used for the old mobile app based on phonegap/ionic/angular